LIMS2DB/classes.py

```diff
@@ -878,6 +878,17 @@ def get_samples(self):
 
             self.get_initial_qc(sample)
             self.get_library_preps(sample)
+            self.get_current_step_sample(sample)
+    
+    def get_current_step_sample(self, sample):
+        """ Get the current steps a sample is in"""
+        sample_in_steps = get_currentsteps_protocol_for_sample(self.session, sample.sampleid)
+        if sample_in_steps:
+            current_steps = {}
+            for step in sample_in_steps:
+                step_details = get_protocolstep_details(self.session, step[0])[0]
+                current_steps[step_details[0]] = {"protocol_name": step_details[1], "is_qc_protocol": step_details[2]}
```

Comment:
Do we need to check whether step_details is not empty list, it would raise an exception in that case.

Reply:
If a sample is assigned to a step in LIMS, the definition for that step should also exist. If we get an empty list, then there would be something very wrong in LIMS.